src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -133,6 +124,28 @@ StandardRuleDataflowLinkOptions GetProjectBuildOptions()
                 {
                     return DataflowOption.WithRuleNames(_applyChangesToWorkspaceContext.Value.GetProjectBuildRules());
                 }
+
+                Task OnEvaluationUpdateAsync(IProjectVersionedValue<(ConfiguredProject ActiveConfiguredProject, IProjectSubscriptionUpdate ProjectUpdate, IProjectSubscriptionUpdate SourceItemsUpdate)> e)
+                {
+                    return OnProjectChangedAsync(
+                        _evaluationProgressRegistration,
+                        e.Value.ActiveConfiguredProject,
+                        lastEvaluationContextState,
+                        e.DataSourceVersions,
+                        hasChange: () => e.Value.ProjectUpdate.ProjectChanges.HasChange() || e.Value.SourceItemsUpdate.ProjectChanges.HasChange(),
```

Comment:
Can you do a trace of this after change? I'd like to have a look at capture/delegate allocations in particular.

Reply:
Actually I think I can eliminate the allocations here altogether by passing what we need on the stack.

Reply:
Those allocations are removed with c3f9af13eb1b53674797944e19081d21e6ec953f.